File: server/app/schemas/article.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ArticleListItem(BaseModel):
    """文章列表项(不含正文)。"""

    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    summary: str = ""
    cover_url: str = ""
    category: str = "general"
    view_count: int = 0
    sort_order: int = 0
    created_at: datetime


class ArticleOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    summary: str = ""
    cover_url: str = ""
    content: str
    category: str = "general"
    view_count: int = 0
    sort_order: int = 0
    created_at: datetime
    updated_at: datetime


class ArticleViewOut(BaseModel):
    article_id: int
    view_count: int
    viewed: bool

File: mini-program/src/api/article.ts
// 文章相关接口
import { http } from './request'

export interface ArticleListItem {
  id: number
  title: string
  summary: string
  cover_url: string
  category: string
  view_count: number
  sort_order: number
  created_at: string
}

export interface ArticleDetail extends ArticleListItem {
  content: string
  updated_at: string
}

export interface ArticleViewResult {
  article_id: number
  view_count: number
  viewed: boolean
}

export function getArticles() {
  return http.get<ArticleListItem[]>('/articles')
}

export function getArticle(id: number) {
  return http.get<ArticleDetail>(`/articles/${id}`)
}

export function reportArticleView(article_id: number) {
  return http.post<ArticleViewResult>(`/articles/${article_id}/view`, {})
}